language/move-binary-format/src/check_bounds.rs

```diff
@@ -516,6 +513,20 @@ impl<'a> BoundsChecker<'a> {
         Ok(())
     }
 
+    fn check_code_unit_bounds_impl_opt<T, I>(
+        &self,
+        pool: &Option<&[T]>,
+        idx: I,
+        bytecode_offset: usize,
+    ) -> PartialVMResult<()>
+    where
+        I: ModuleIndex,
+    {
+        pool.map_or(Ok(()), |p| {
```

Comment:
Initially I was thinking a pattern match would be better here, but this is probably good.

Thanks for the change!